security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,131 @@
+--- Makefile.orig	2021-09-09 19:58:05.000000000 +1000
++++ Makefile	2023-06-12 13:16:16.000000000 +1000
+@@ -23,10 +23,10 @@
+ ID := $(NAME)-$(VERSION)
+ 
+ DESTDIR :=
+-PREFIX := $(DESTDIR)/usr
++PREFIX := $(DESTDIR)/opt/local
+ BINDIR := $(PREFIX)/bin
+ MANDIR := $(PREFIX)/share/man
+-ETCDIR := $(DESTDIR)/etc
++ETCDIR := $(PREFIX)/etc
+ 
+ POD2MAN := pod2man
+ POD2HTML := pod2html
+@@ -59,13 +59,15 @@
+ check: test
+ 
+ install: man
+-	[ -d $(ETCDIR)/$(SSHDOERS).d ] || mkdir -m 755 $(ETCDIR)/$(SSHDOERS).d
+-	[ -f $(ETCDIR)/$(SSHDOERS) ] || install -o root -g root -m 644 $(SSHDOERS) $(ETCDIR)
+-	[ -f $(ETCDIR)/$(SSHDO_BANNER) ] || install -o root -g root -m 644 $(SSHDO_BANNER) $(ETCDIR)
+-	install -o root -g root -m 755 $(BIN) $(BINDIR)
++	[ -d $(ETCDIR)/$(SSHDOERS).d ] || mkdir -p -m 755 $(ETCDIR)/$(SSHDOERS).d
++	[ -f $(ETCDIR)/$(SSHDOERS) ] || install -o root -g wheel -m 644 $(SSHDOERS) $(ETCDIR)
++	[ -f $(ETCDIR)/$(SSHDO_BANNER) ] || install -o root -g wheel -m 644 $(SSHDO_BANNER) $(ETCDIR)
++	install -o root -g wheel -m 755 $(BIN) $(BINDIR)
+ 	[ -z "`which python3`" -a -n "`which python`" ] && sed 's/env python3$$/env python/' < $(BIN) > $(BINDIR)/$(BIN) || true
```

Comment:
Thanks. Why would it be better? There are no non-standard python modules needed. I think it's better to work with whatever version of python the local system already has. There's no need to download and install a different version. I also think it it's better to reduce the prerequisites as much as possible.

But if there's a good reason to require a particular macports python version, which one to pick? It works with 2.6, 2.7, and 3.3+. Will I have to keep changing it every so often? Or is there a port that means the current default version of python3?

Reply:
for example because older OSes do not have "python3" and newer ones eventually will not have Python pre-installed if I am not mistaken. 

Typically, one uses the default as is set in the "python" PG. that version is incremented every year on January 1.

Reply:
Thanks. It works with python 2 or 3. But if there isn't going to be a system python3 in future, that would be a problem. I've changed it to depend on port python311. That seems to be the current one. I'm not using the python port group because it seems to assume that setup.py will be used for installation, but this program uses a Makefile.

Reply:
Thanks. How do I do that?

Reply:
If the user has already selected some version of python3 (which I expect would usually be the case), I don't think a port should go against the user's wishes and change the selected python3 version to some other version. That seems rude. But perhaps I'm misinterpreting, and that's not what you are saying. If the port is required to depend on a particular version of python, and it's not allowed to assume that the user has selected their preferred version of python3, maybe I should just create a patch to change the python3 to python3.11. Would that be an accpetable approach?

Reply:
I've just done it that way. Now it won't matter which version of python3 is selected or whether any version is selected.